demos/cli/cli_uart_demo.c

```diff
@@ -0,0 +1,290 @@
+    configASSERT( pcWriteBuffer );
+
+    if( lParameterNumber == 0 )
+    {
+        /* The first time the function is called after the command has been
+         * entered just a header string is returned. */
+        sprintf( pcWriteBuffer, "The parameters were:\r\n" );
+
+        /* Next time the function is called the first parameter will be echoed
+         * back. */
+        lParameterNumber = 1L;
+
+        /* There is more data to be returned as no parameters have been echoed
+         * back yet. */
+        xReturn = pdPASS;
+    }
+    else
+    {
+        /* Obtain the parameter string. */
+        pcParameter = FreeRTOS_CLIGetParameter
+                      (
+            pcCommandString,        /* The command string itself. */
+            lParameterNumber,       /* Return the next parameter. */
+            &lParameterStringLength /* Store the parameter string length. */
+                      );
+
+        if( pcParameter != NULL )
+        {
+            /* Return the parameter string. */
+            memset( pcWriteBuffer, 0x00, xWriteBufferLen );
+            sprintf( pcWriteBuffer, "%d: ", lParameterNumber );
+            strncat( pcWriteBuffer, pcParameter, lParameterStringLength );
+            strncat( pcWriteBuffer, "\r\n", strlen( "\r\n" ) );
+
+            /* There might be more parameters to return after this one. */
+            xReturn = pdTRUE;
+            lParameterNumber++;
+        }
+        else
+        {
+            /* No more parameters were found.  Make sure the write buffer does
+             * not contain a valid string. */
+            pcWriteBuffer[ 0 ] = 0x00;
+
+            /* No more data to return. */
+            xReturn = pdFALSE;
+
+            /* Start over the next time this command is executed. */
+            lParameterNumber = 0;
+        }
+    }
+
+    return xReturn;
+}
+
+static BaseType_t prvTaskStatsCommand( char * pcWriteBuffer,
+                                       size_t xWriteBufferLen,
+                                       const char * pcCommandString )
+{
+    const char * const pcHeader = "Task          State  Priority  Stack	#\r\n************************************************\r\n";
+
+    /* Remove compile time warnings about unused parameters, and check the
+     * write buffer is not NULL.  NOTE - for simplicity, this example assumes the
+     * write buffer length is adequate, so does not check for buffer overflows. */
+    ( void ) pcCommandString;
+    ( void ) xWriteBufferLen;
+    configASSERT( pcWriteBuffer );
+
+    /* Generate a table of task stats. */
+    strcpy( pcWriteBuffer, pcHeader );
+    vTaskList( pcWriteBuffer + strlen( pcHeader ) );
+
+    /* There is no more data to return after this single string, so return
+     * pdFALSE. */
+    return pdFALSE;
+}
+
+void vRegisterCLICommands( void )
+{
+    /* Register all the command line commands defined immediately above. */
```

Comment:
Not immediately above but at the top of the file.

Reply:
Fixed.